modules/features2d/src/blobdetector.cpp

```diff
@@ -71,6 +71,37 @@ class CV_EXPORTS_W SimpleBlobDetectorImpl : public SimpleBlobDetector
   virtual void read( const FileNode& fn ) CV_OVERRIDE;
   virtual void write( FileStorage& fs ) const CV_OVERRIDE;
 
+  void setParams(const SimpleBlobDetector::Params& _params ) CV_OVERRIDE {
+    SimpleBlobDetectorImpl::validateParameters(_params);
+    params = _params;
+  }
+
+  SimpleBlobDetector::Params getParams() const CV_OVERRIDE { return params; }
+
+  static void validateParameters(const SimpleBlobDetector::Params& p)
+  {
+      if (p.thresholdStep <= 0)
+          CV_Error(Error::StsBadArg, "thresholdStep>0");
+
+      if (p.minThreshold > p.maxThreshold || p.minThreshold <= 0)
```

Comment:
@augustinmanecy Should that be `p.minThreshold < 0`?

Some of my own internal tests are now failing against OpenCV master, and I think this is why.

Reply:
https://github.com/opencv/opencv/pull/23036